pipeline/state_duration.go

```diff
@@ -0,0 +1,58 @@
+package pipeline
+
+import (
+	"time"
+
+	"github.com/influxdata/kapacitor/tick/ast"
+)
+
+// Compute the duration of a given state.
+// The state is defined via a lambda expression. For each consecutive point for
+// which the expression evaluates as true, the state duration will be
+// incremented by the duration between points. When a point evaluates as false,
+// the state duration is reset to 0.
+//
+// The state duration will be added as an additional field to each point. If the
+// expression evaluates as false, the field will not be added. If the expression
```

Comment:
Maybe someone wants alert when the duration is less than X. Silly example, but maybe a beer factory wants to turn on an indicator light when a keg is full, but hasn't sat long enough to finish fermenting (light on means "fermentation in progress"). If it's 0 when expression evaluates as false (keg is empty), the light would be on.

Or maybe someone wants to have a critical state when the expression is false, warning state when it's less than X, and OK state when it's greater than X.

[Defensive programming](https://en.wikipedia.org/wiki/Defensive_programming). I just want it to be able to handle all use cases, even the ones I can't think of.